workspaces/adr/plugins/adr-backend/src/service/router.test.ts

```diff
@@ -228,4 +232,72 @@ describe('createRouter', () => {
       expect(fileTwoBody.data).toBe(testFileTwoContent);
     });
   });
+
+  describe(`GET ${imageEndpointName}`, () => {
+    it('returns bad request (400) when no url is provided', async () => {
+      const urlNotSpecifiedRequest = await request(app).get(imageEndpointName);
+      const urlNotSpecifiedStatus = urlNotSpecifiedRequest.status;
+      const urlNotSpecifiedMessage = urlNotSpecifiedRequest.body.message;
+
+      const urlNotFilledRequest = await request(app).get(
+        `${imageEndpointName}?url=`,
+      );
+      const urlNotFilledStatus = urlNotFilledRequest.status;
+      const urlNotFilledMessage = urlNotFilledRequest.body.message;
+
+      const expectedStatusCode = 400;
+      const expectedErrorMessage = 'No URL provided';
+
+      expect(urlNotSpecifiedStatus).toBe(expectedStatusCode);
+      expect(urlNotSpecifiedMessage).toBe(expectedErrorMessage);
+
+      expect(urlNotFilledStatus).toBe(expectedStatusCode);
+      expect(urlNotFilledMessage).toBe(expectedErrorMessage);
+    });
+
+    it('returns bad request (400) when unsupported image format is provided', async () => {
+      const urlNotFilledRequest = await request(app).get(
+        `${imageEndpointName}?url=testImage.txt`,
+      );
+      const urlNotFilledStatus = urlNotFilledRequest.status;
+      const urlNotFilledMessage = urlNotFilledRequest.body.message;
+
+      const expectedStatusCode = 400;
+      const expectedErrorMessage = 'Image type txt is not supported';
+
+      expect(urlNotFilledStatus).toBe(expectedStatusCode);
+      expect(urlNotFilledMessage).toBe(expectedErrorMessage);
+    });
+
+    it('returns the correct image when reading a url', async () => {
+      const urlToProcess = 'testImage.png';
+      const imageTypeMap: Record<string, string> = {
+        png: 'image/png',
+        jpg: 'image/jpeg',
+        jpeg: 'image/jpeg',
+        gif: 'image/gif',
+        svg: 'image/svg+xml',
+        webp: 'image/webp',
+      };
+      const imageResponse = await request(app).get(
+        `${imageEndpointName}?url=${urlToProcess}`,
+      );
+      const imageStatus = imageResponse.status;
+      const imageData = imageResponse.body;
+      const imageError = imageResponse.error;
+      const imageType = urlToProcess.match(/\.([a-z0-9]+)(\?.*)?$/i);
+      let contentType;
+      if (imageType) {
+        contentType = imageTypeMap[imageType[1].toLowerCase()];
+      }
```

Comment:
We probably can hard code this instead of dynamically calculating this since we're in a controlled test scenario right?

Reply:
Yes, you are 100% right.
I hard coded it :)